mytardis_rocrate_builder/rocrate_dataclasses/rocrate_dataclasses.py

```diff
@@ -34,177 +41,239 @@ class DataClassification(Enum):
-    def id(self) -> str | int | float:
-        """Retrieve first ID to act as RO-Crate ID"""
-        return self.identifiers[0]
+    def __post_init__(self) -> None:
+        object.__setattr__(
+            self, "identifier", gen_uuid_id(MYTARDIS_NAMESPACE_UUID, self.name)
+        )
 
 
-@dataclass
-class Group:
-    """Dataclass to hold the details of a group for RO-Crate
-    
+@dataclass(kw_only=True)
+class User(Person):  # pylint: disable=too-many-instance-attributes
+    """Dataclass to extend Person as a Django user in MyTardis.
+        Primarily used to link people to Groups if needed,
+        But capable of storing all information
     Attr:
-        name (str): The group name of the group
-    """
-    name: str
-    schema_type: List[str] = ["Audience", "Thing"]
-    @property
-    def id(self) -> str:
-        """Return the group name as the RO-Crate ID"""
-        return self.name
-
-@dataclass
-class BaseObject(ABC):
-    """Abstract Most basic object that can be turned into an RO-Crate entity"""
+        groups (Optional[List[Group]]): All groups this user belongs to
+        isDjangoAccount (Optional[bool]): was this user created in Django
+        permissions (Optional[Dict[str:Any]]): all permissions held by this user account
+        is_staff (Optional[bool]): is this user a staff/admin account
+        hashed_password (Optional[str]): the hashed password of the user
+            DO NOT STORE RAW PASSWORDS
+        is_active (Optional[bool]): is this an active user account
+        is_superuser (Optional[bool]): is this user a superuser? (should we store this?)
+        last_login (Optional[datetime]): last login date of this user
+        date_joined (Optional[datetime]): when did this user join the service
 
-    name: str
-
-
-@dataclass
-class MTMetadata(BaseObject):
-    """Concrete Metadata class for RO-crate
-    Contains all information to store or recreate MyTardis metadata.
-    Used as backup and recovery option.
-
-    creates an RO-Crate JSON-LD entity matching this schema
-    "@id": string - unique ID in the RO-Crate,
-    "@type": string = "MyTardis-Metadata_field" - RO-Crate type,
-    "name": string - name of the metadata in MyTardis,
-    "value": string | Any - Metadata value in my tardis,
-    "mt-type": string - Metadata type as recorded in MyTardis,
-    "mt-schema": string - the object schema in MyTardis that applies to this metadata record
-    "sensitive": bool - Metadata ,
-    "parents": List[string] - list of ids for all the parents
-
-    Attr:
-        experiment (str): An identifier for an experiment
     """
-    ro_crate_id: str
-    value: str
-    mt_type: str
-    mt_schema: str
-    sensitive: bool
-    parents: List[str] | None
-
 
-@dataclass
-class ContextObject(BaseObject):
+    groups: Optional[List[Group]] = None
+    is_django_account: bool = False
+    permissions: Optional[Dict[str, str]] = None
+    is_staff: Optional[bool] = None
+    hashed_password: Optional[str] = None
```

Comment:
See previous comment about storing hashed password. I'm not sure that we need to do so and wonder if the risk of doing this outweighs the convenience. Anyone can take a look at the Django code base to see the salt for the salted hash used for the password. Possibly being paranoid but....
I think its worth thinking about the implications of not storing the hashed password.